lib/queue_classic.rb

```diff
@@ -49,17 +49,19 @@ def self.has_connection?
   end
 
   def self.default_conn_adapter
-    return @conn_adapter if defined?(@conn_adapter) && @conn_adapter
-    if rails_connection_sharing_enabled?
-      @conn_adapter = ConnAdapter.new(ActiveRecord::Base.connection.raw_connection)
+    t = Thread.current
+    return t[:conn_adapter] if t[:conn_adapter]
```

Comment:
as `Thread.current` is global state, we should go for a key which has less conflict potential. What about `qc_conn_adapter`?